--- yarn/yarn.lock	2022-10-12 10:35:35.000000000 -0700
+++ yarn/new-lock	2022-10-12 12:51:22.000000000 -0700
@@ -43,7 +43,7 @@
   integrity sha512-p+LQN9O39+rZuOAyc6BzyVGvdEKo+v+XmtdeyZsZpfj4xuOLtsEptW1w6cUD439u0YcPknuccGq1MQ0lXQ6Xuw==
 
 turbo@^1.5.6:
-  version "1.5.6"
+  version "1.5.7"
   resolved "https://registry.yarnpkg.com/turbo/-/turbo-1.5.6.tgz#7dac8db14b2452afa45c57c050ff19989d4ab074"
   integrity sha512-xJO/fhiMo4lI62iGR9OgUfJTC9tnnuoMwNC52IfvvBDEPlA8RWGMS8SFpDVG9bNCXvVRrtUTNJXMe6pJWBiOTA==
   optionalDependencies:
