From 2a6f7f5af9b82a568fa5ca8ad4d98edc910c68ed Mon Sep 17 00:00:00 2001
From: Artur Troian <troian.ap@gmail.com>
Date: Fri, 12 Nov 2021 18:45:09 -0500
Subject: [PATCH] fix(wrapper): inject libc++ when sdk version >= 11.1

fixes #313

Signed-off-by: Artur Troian <troian.ap@gmail.com>
---
 wrapper/target.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/wrapper/target.cpp b/wrapper/target.cpp
index 82bf65c..a81ce97 100644
--- a/wrapper/target.cpp
+++ b/wrapper/target.cpp
@@ -741,6 +741,10 @@ bool Target::setup() {
           (stdlib == StdLib::libstdcxx && usegcclibs)) {
         fargs.push_back("-nostdinc++");
         fargs.push_back("-Qunused-arguments");
+
+        if ((SDKOSNum >= OSVersion(11, 1)) && (stdlib == StdLib::libcxx)) {
+          fargs.push_back("-lc++");
+        }
       }
 
       if (stdlib == StdLib::libstdcxx && usegcclibs && targetarch.size() < 2 &&
-- 
2.30.1 (Apple Git-130)

