diff --git a/node_modules/@remix-run/dev/compiler.js b/node_modules/@remix-run/dev/compiler.js
index 902b251..d5b4240 100644
--- a/node_modules/@remix-run/dev/compiler.js
+++ b/node_modules/@remix-run/dev/compiler.js
@@ -211,6 +211,10 @@ async function watch(config$1, {
     toWatch.push(config$1.serverEntryPoint);
   }
 
+  if(config$1.serverDependenciesToBundle) {
+    toWatch.push(config$1.serverDependenciesToBundle.map(moduleName => require.resolve(moduleName).replace(/[^\/]*$/,"**")));
+  }
+
   let watcher = chokidar__default["default"].watch(toWatch, {
     persistent: true,
     ignoreInitial: true,
