--- a/capnp/compat/json.capnp	2021-10-21 14:59:51.244324279 -0400
+++ b/capnp/compat/json.capnp	2021-10-21 15:01:09.543760012 -0400
@@ -28,7 +28,7 @@
     null @0 :Void;
     boolean @1 :Bool;
     number @2 :Float64;
-    string @3 :Text;
+    string @3 :Text $Go.name("string_");
     array @4 :List(Value);
     object @5 :List(Field);
     # Standard JSON values.
--- a/capnp/schema.capnp	2021-10-21 15:03:20.221774538 -0400
+++ b/capnp/schema.capnp	2021-10-21 15:04:20.870019327 -0400
@@ -82,7 +82,7 @@

     file @6 :Void;

-    struct :group {
+    struct :group $Go.name("structNode") {
       dataWordCount @7 :UInt16;
       # Size of the data section, in words.

@@ -336,7 +336,7 @@
       typeId @15 :Id;
       brand @21 :Brand;
     }
-    struct :group {
+    struct :group $Go.name("structType") {
       typeId @16 :Id;
       brand @22 :Brand;
     }
@@ -450,7 +450,7 @@
     list @14 :AnyPointer;

     enum @15 :UInt16;
-    struct @16 :AnyPointer;
+    struct @16 :AnyPointer $Go.name("structValue");

     interface @17 :Void;
     # The only interface value that can be represented statically is "null", whose methods always
--- a/capnp/persistent.capnp	2021-10-21 15:12:33.501158612 -0400
+++ b/capnp/persistent.capnp	2021-10-21 15:13:31.930030219 -0400
@@ -108,7 +108,7 @@
   }
 }

-annotation persistent(interface, field) :Void;
+annotation persistent(interface, field) :Void $Go.name("PersistentAnnotation");
 # Apply this annotation to interfaces for objects that will always be persistent, instead of
 # extending the Persistent capability, since the correct type parameters to Persistent depend on
 # the realm, which is orthogonal to the interface type and therefore should not be defined
