--- pnpm/pnpm-lock.yaml	2022-10-12 13:33:53.000000000 -0700
+++ pnpm/new	2022-10-12 13:38:25.000000000 -0700
@@ -6,7 +6,7 @@
     specifiers:
       turbo: ^1.5.6
     dependencies:
-      turbo: 1.5.6
+      turbo: 1.5.7
 
   apps/a:
     specifiers:
@@ -79,7 +79,7 @@
     dev: false
     optional: true
 
-  /turbo/1.5.6:
+  /turbo/1.5.7:
     resolution: {integrity: sha512-xJO/fhiMo4lI62iGR9OgUfJTC9tnnuoMwNC52IfvvBDEPlA8RWGMS8SFpDVG9bNCXvVRrtUTNJXMe6pJWBiOTA==}
     hasBin: true
     requiresBuild: true
