--- bun.lock	2024-01-01 00:00:00.000000000 -0700
+++ new	2024-01-01 00:02:00.000000000 -0700
@@ -22,7 +22,7 @@
   },
   "packages": {
     "function-bind": [
-      "function-bind@1.1.2",
+      "function-bind@1.1.2",
       {},
       "sha512-yIovAzMX49sF8Yl58fSCWJ5svSLuaibPxXQJFLmBObTuCr0Mf1KiPopGM9NiFjiYBCbfaa2Fh6breQ6ANVTI0A=="
     ],
@@ -32,7 +32,7 @@
       "sha512-l3LCuF6MgDNwTDKkdYGEihYjt5pRPbEg46rtlmnSPlUbgmB8LOIrKJbYYFBSbnPaJexMKtiPO8hmeRjRz2Td+A=="
     ],
     "turbo": [
-      "turbo@1.5.6",
+      "turbo@1.5.7",
       {},
       "sha512-xJO/fhiMo4lI62iGR9OgUfJTC9tnnuoMwNC52IfvvBDEPlA8RWGMS8SFpDVG9bNCXvVRrtUTNJXMe6pJWBiOTA=="
     ]
